File: scrape_exact.py
import requests
from bs4 import BeautifulSoup
import json
import time

all_items = []
headers = {'User-Agent': 'Mozilla/5.0'}

for page in range(1, 9):
    url = f"https://centrerabelaislyon.fr/publications/page/{page}/" if page > 1 else "https://centrerabelaislyon.fr/publications/"
    try:
        r = requests.get(url, headers=headers, timeout=10)
        soup = BeautifulSoup(r.content, 'html.parser')
        articles = soup.find_all('article')
        print(f"Page {page}: found {len(articles)} articles")
        
        for article in articles:
            # Title
            title_elem = article.find(class_="entry-title")
            if not title_elem:
                continue
            title = title_elem.text.strip()
            
            # Thumbnail
            thumb = ""
            media_elem = article.find(class_="entry-media")
            if media_elem:
                img = media_elem.find('img')
                if img and img.get('src'):
                    thumb = img.get('src')
            
            # Author photo
            author_photo = ""
            author_section = article.find(class_="author-avatar") # guessing class
            if not author_section:
                # search for any IMG inside text blocks that looks like an author photo
                # in screenshot 1: "AUTHORS" -> circle photo
                # "wp-image-something" or similar
                imgs = article.find_all('img')
                for i in imgs:
                    src = i.get('src', '')
                    if 'team' in src or 'Image-37' in src or 'Avatar' in src or 'post' in src:
                        # try to identify the author avatar from multiple imgs
                        pass
                # The author photo in Screenshot 1 was actually inside the article block!
                author_div = article.find('div', class_='author-info') or article.find('div', class_='post-author') or article.find(string="AUTHORS")
                if author_div:
                    # traverse up then down
                    p = author_div.parent if hasattr(author_div, 'parent') else author_div
                    aimg = getattr(p, 'find', lambda x: None)('img')
                    if aimg:
                        author_photo = aimg.get('src')
            
            # Just grab all images in the article block and figure it out
            all_imgs = [i.get('src') for i in article.find_all('img')]
            
            all_items.append({
                "title": title,
                "thumbnail": thumb,
                "all_imgs": all_imgs
            })
    except Exception as e:
        print(f"Error {page}: {e}")

with open("exact_mapping.json", "w", encoding="utf-8") as f:
    json.dump(all_items, f, indent=2)
print(f"Saved {len(all_items)} entries")

File: scripts/scrape_exact.js
const axios = require('axios');
const cheerio = require('cheerio');
const fs = require('fs');

async function scrapeListingPage(pageNumber) {
    const url = pageNumber === 1 
        ? 'https://centrerabelaislyon.fr/publications/' 
        : `https://centrerabelaislyon.fr/publications/page/${pageNumber}/`;
        
    try {
        const res = await axios.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } });
        const $ = cheerio.load(res.data);
        
        const pageItems = [];
        
        // Find each publication block. On the original site, they reside within an <article> usually
        $('article.post').each((i, el) => {
            const title = $(el).find('.entry-title').text().trim();
            
            // The thumbnail is usually inside .entry-media or similar
            const thumbImg = $(el).find('.entry-media img').attr('src');
            
            // The author photo might be inside another block, let's look for anything that resembles "authors"
            // If the original site doesn't have authors on the listing page, then the thumbImg is enough
            
            if (title && thumbImg) {
                pageItems.push({
                    title,
                    thumbnail: thumbImg
                });
            }
        });
        
        return pageItems;
    } catch (e) {
        console.error(`Page ${pageNumber} failed: ${e.message}`);
        return [];
    }
}

(async () => {
    let allData = [];
    for (let i = 1; i <= 8; i++) {
        console.log(`Scraping page ${i}...`);
        const items = await scrapeListingPage(i);
        allData = allData.concat(items);
    }
    
    fs.writeFileSync('exact_mapping.json', JSON.stringify(allData, null, 2));
    console.log(`Successfully mapped ${allData.length} true thumbnails from the original site.`);
})();
